src/trace.c

```diff
@@ -0,0 +1,115 @@
+// This file is a part of Julia. License is MIT: https://julialang.org/license
+//
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "julia.h"
+#include "julia_internal.h"
+
+#ifdef ENABLE_TRACING
+
+#include <stdlib.h>
+#include "threading.h"
+
+typedef struct _jl_tracepkt_t {
+    int16_t tid;
+    uint64_t timestamp;
+    uint64_t event;
+
```

Comment:
You could pack more into a single UDP packet by coalescing the common data:
```suggestion
    uint16_t tid;
    uint32_t pid;
    uint64_t timestamp0;
    struct {
        uint16_t timestamp;
        uint16_t event;
    } data[(1472 - 128) / 32];
```